### DISCUSSION TITLE
Set stop duration in via points

### DISCUSSION THREAD

**Discussion by devemux86** [devemux86/cruiser#573] [Q&A] [answered]
We could add an option to define stop duration in the via points during route planning.

So if we want to stop our journey for some time at via points,
we can know better when we will arrive at the destination or other via points.

The stop duration will be used in all time calculations in route planning and navigation,
such as remaining time and arrival time.

**Comment by devemux86** [comment]
BaseCamp seems to have this feature?

**Reply by 0709wiwiwi** [reply]
Plotaroute web offers this and maybe also some others (web) routers too ?   
I did not check but it seems also Furkot and MRA do offer this (Garmin) feature  ?

**Reply by WolfgangZ42** [reply]
Please tell me Willi: Is there any navigation software you did not install? 🤔😏
I've never heard of Plotaroute, Furkot and some others you mentioned here already ...

**Comment by devemux86** [comment]
@0709wiwiwi

We should think about how we can store the stop duration of via points in the [GPX navigation track format](https://github.com/devemux86/cruiser/discussions/250).

**Comment by MotoUKRider** [comment]
Yes, BaseCamp indeed does this - I've never used it, but it could be handy for us Brits who like our roadside cup of tea at precisely 11:30am.

![basecamp-layover](https://github.com/devemux86/cruiser/assets/125696109/5d444061-5975-4d74-ac00-b1cf9a07934d)

**Comment by airald** [comment]
This is a Garmin GPX file (generated by Furkot)

    <rtept lat="41.950042" lon="2.251725">
      <time>2024-07-01T08:04:26.000Z</time>
      <sym>Bar</sym>
      <extensions>
        <trp:ViaPoint>
          <trp:DepartureTime>2024-07-01T14:34:26.000Z</trp:DepartureTime>
          <trp:StopDuration>PT6H30M</trp:StopDuration>                                    <<<< --- STOP DURATION
          <trp:ArrivalTime>2024-07-01T08:04:26.000Z</trp:ArrivalTime>
          <trp:CalculationMode>FasterTime</trp:CalculationMode>
          <trp:ElevationMode>Standard</trp:ElevationMode>
        </trp:ViaPoint>
        <gpxx:RoutePointExtension>
          <gpxx:Subclass>000000000000FFFFFFFFFFFFFFFFFFFFFFFF</gpxx:Subclass>
          <gpxx:rpt lat="41.950042" lon="2.251725"></gpxx:rpt>
          <gpxx:rpt lat="41.950865" lon="2.251501"></gpxx:rpt>
          <gpxx:rpt lat="41.950884" lon="2.251550"></gpxx:rpt>
       ...
          <gpxx:rpt lat="41.999965" lon="2.269515"></gpxx:rpt>
          <gpxx:rpt lat="41.999856" lon="2.269554"></gpxx:rpt>
          <gpxx:rpt lat="41.999816" lon="2.269525"></gpxx:rpt>
          <gpxx:rpt lat="41.999789" lon="2.269499"></gpxx:rpt>
          <gpxx:rpt lat="41.999789" lon="2.269499"></gpxx:rpt>
          <gpxx:rpt lat="41.999789" lon="2.269499">
            <gpxx:Subclass>000000000000FFFFFFFFFFFFFFFFFFFFFFFF</gpxx:Subclass>
          </gpxx:rpt>
        </gpxx:RoutePointExtension>
      </extensions>
    </rtept>

If I do the same test but exporting as standard GPX file, the stop duration is not saved.

**Comment by 0709wiwiwi** [comment]
**CRUISER** can offer this KISS. 
Add the **expected** track start date & start time.
Add stop time (seconds) in the (color) via points.

`<extensions><xsi:gpx><stop>1200</stop><color>#FF000000</color></xsi:gpx></extensions> `

The via points ETA and final ETA are attached in the **trackpoint timestamps.**

![via_stop_1200](https://github.com/user-attachments/assets/4e7266ba-7f44-4b7a-a697-eed46333938a)

Trough the trackpoint timestamps, notice the stops by zero speed. The avg speed drops accordingly.

**Comment by devemux86** [comment, marked as answer]
I am thinking of something simple, like an extra field in via points to define the stop duration.
The route timings will then be updated automatically, offline without recalculating the route.

The result will be visible in all information screens and navigation.
Maybe we'll consider it in the future.

**Comment by 0709wiwiwi** [comment]
Q: Where does the difference in via ypoint times between the roadbook and the navigation system come from?

Example file 1. Has a single tkrpt type via stop notation. (Timestamps are  +/-  as a Plotaroute track export)
Example file 2. Has a trkpt type via stop notation plus next at the same position an extra trkpt named via go.

The time to the destination does match the roadbook, for some reason the time shown to the via point deviates
<img width="607" height="574" alt="Via stop and go" src="https://github.com/user-attachments/assets/9be4c0f6-f86c-43f6-8f2f-2452717a44a6" />
[1_Via stop_navtrk.gpx.txt](https://github.com/user-attachments/files/30869316/1_Via.stop_navtrk.gpx.txt)
[2_Via stop_go_navtrk.gpx.txt](https://github.com/user-attachments/files/30852570/2_Via.stop_go_navtrk.gpx.txt)




.

**Reply by devemux86** [reply]
There is not any difference, in navigation you selected to see the distance / time to the destination.
- Tap the distance panel at the right (with the flag) to switch between destination and next via points.

- Long press the distance panel at the right (with the flag) to select the next via point to see its info.

The road book in navigation and route planning is explained in its thread:

- https://github.com/devemux86/cruiser/discussions/473

---

Although it's not clear, what do you mean?
Please attach screenshots that are easy to see, not animated images that are fast and difficult to see.

**Reply by 0709wiwiwi** [reply]
The roadbook read out does take the imported track timestamps into account.
In the navigation, it is apparently recalculated by the tracklenght avg speed ?
It certainly does not match each other.

**Reply by devemux86** [reply]
The navigation road book shows the real time until the next waypoints.
Like navigation panels, it updates as we move.

**Reply by 0709wiwiwi** [reply]
The roadbook says 10 minutes to the via point Kapel.
But it becomes 18 minutes at the navigation start.

**Reply by devemux86** [reply]
Your 1st screenshot shows the **static** road book in the routing menu.
That information is provided by the calculated / imported route.

Your 2nd screenshot shows the navigation panels that are updated in **real time**.
The navigation road book is also updated in **real time**.

**Reply by 0709wiwiwi** [reply]
I am at the starting point and haven't even set off yet; surely both should be identical here? No ?

**Reply by devemux86** [reply]
No, because navigation data are calculated differently.
Real time calculations can not use the static route info.

**Reply by devemux86** [reply]
Also, on normal longer routes, such minor differences are not significant.

**Reply by 0709wiwiwi** [reply]
> Real time calculations can not use the static route info.

Roger

**Reply by devemux86** [reply]
It has nothing to do with speed, it's technical details.

**Reply by 0709wiwiwi** [reply]
> Also, on normal longer routes, such minor differences are not significant.

Anyway it is fine that a (one hour) stop is shown in the roadbook results.  👍

**Reply by devemux86** [reply]
There is no stop duration option in via points.
We will see in the future based on user interest.
